backend/compact-connect/lambdas/staff-users/tests/function/test_data_model/test_client.py

```diff
@@ -0,0 +1,442 @@
+            compact='aslp',
+            user_id=user_id,
+            compact_action_removals={'admin'},
+            jurisdiction_action_additions={
+                'oh': {'write', 'admin'}
+            }
+        )
+
+        self.assertEqual(user_id, resp['userId'])
+        self.assertEqual(
+            {
+                'actions': {'read'},
+                'jurisdictions': {
+                    'oh': {'write', 'admin'}
+                }
+            },
+            resp['permissions']
+        )
+        # Checking that we're getting the whole object, not just changes
+        self.assertFalse(
+            {'type', 'userId', 'compact', 'attributes', 'permissions', 'dateOfUpdate'} - resp.keys()
+        )
+
+    def test_update_user_attributes(self):
+        # The sample user looks like board staff in aslp/oh
+        user_id = UUID(self._load_user_data())
+
+        from data_model.client import UserClient
+
+        client = UserClient(self.config)
+
+        resp = client.update_user_attributes(
+            user_id=user_id,
+            attributes={
+                'givenName': 'Bob',
+                'familyName': 'Smith'
+            }
+        )
+        self.assertEqual(1, len(resp))
+        user = resp[0]
+
+        self.assertEqual(user_id, user['userId'])
+        self.assertEqual(
+            {
+                'givenName': 'Bob',
+                'familyName': 'Smith',
+                'email': 'justin@example.org'
+            },
+            user['attributes']
+        )
+        # Checking that we're getting the whole object, not just changes
+        self.assertFalse(
+            {'type', 'userId', 'compact', 'attributes', 'permissions', 'dateOfUpdate'} - user.keys()
+        )
+
+    def test_create_new_user(self):
+        from data_model.client import UserClient
+
+        client = UserClient(self.config)
+
+        resp = client.create_user(
+            compact='aslp',
+            attributes={
+                'givenName': 'Bob',
+                'familyName': 'Smith',
+                'email': 'bob@example.org'
+            },
+            permissions={
+                'actions': {'read'},
+                'jurisdictions': {
+                    'oh': {'write', 'admin'}
+                }
+            }
+        )
+
+        self.assertEqual(
+            {'type', 'userId', 'compact', 'attributes', 'permissions', 'dateOfUpdate'},
+            resp.keys()
+        )
+        self.assertEqual(
+            {
+                'givenName': 'Bob',
+                'familyName': 'Smith',
+                'email': 'bob@example.org'
+            },
+            resp['attributes']
+        )
+        self.assertEqual(
+            {
+                'actions': {'read'},
+                'jurisdictions': {
+                    'oh': {'write', 'admin'}
+                }
+            },
+            resp['permissions']
+        )
+
+    def test_create_existing_user(self):
```

Comment:
A bit confused re: expected behavior here?

Reply:
In the case that two compact admins invite the same individual to their respective compacts, we want there to only be a single Cognito user created, but two database records for that user (one for each compact). The idea is that the actual creation of a new user vs adding permissions to an existing user should be be invisible to admins across compacts. So what we're looking at here is that we have two sets of permissions (DB records, internally) but that they share the same `userId`.

Reply:
Makes sense. Mind adding a brief comment? I think that's not 100% clear from this. A trimmed down version of that explanation is great.